### PR TITLE
US142821 Add FACE Discussions specific category association action

### PR DESCRIPTION
**Pull request by mostlyfabulous**
[US142821](https://rally1.rallydev.com/#/?detail=/userstory/645735331867&fdp=true): [api] Create choose forum dialog

### CODE REVIEW

File: src/activities/CategoriesEntity.js [hunk 1]
```diff
@@ -165,13 +165,17 @@ export class CategoriesEntity extends Entity {
 		return categoryId !== initialId;
 	}
 
-	async save(category, shouldDeletePreviousCategory = false) {
+	async save(category, shouldDeletePreviousCategory = null) {
```
Comment (mostlyfabulous) [inline src/activities/CategoriesEntity.js:168]: This prevents FACE Discussions from generating `select` & `deselect` actions for saving.

File: src/activities/CategoriesEntity.js [hunk 2]
```diff
@@ -165,13 +165,17 @@ export class CategoriesEntity extends Entity {
...
+			const { action, fields } = this._generateDeletePreviousAndAssociateWithNewCategory(category, shouldDeletePreviousCategory) || {};
+
+			return action && await performSirenAction(this._token, action, fields);
+		}
+
 		if (hasCategoryIdChanged && category.categoryId !== UNSET_CATEGORY_ID) {
-			const { action, fields } = shouldDeletePreviousCategory
-				? this._generateDeletePreviousAndAssociateWithNewCategory(category, shouldDeletePreviousCategory) || {}
-				: this._generateSelectCategoryAction(category) || {};
+			const { action, fields } = this._generateSelectCategoryAction(category) || {};
```
Comment (mostlyfabulous) [inline src/activities/CategoriesEntity.js:178]: Restored back to old implementation.

File: src/activities/CategoriesEntity.js [hunk 3]
```diff
@@ -149,7 +149,7 @@ export class CategoriesEntity extends Entity {
 		const categoryEntity = this._getCategoryById(category.categoryId);
 		if (!categoryEntity) return;
 
-		const action = shouldDeletePreviousCategory && categoryEntity.getActionByName(Actions.activities.categories.deletePreviousAndAssociateWithNewCategory);
+		const action = categoryEntity.getActionByName(Actions.activities.categories.deletePreviousAndAssociateWithNewCategory);
```
Comment (mostlyfabulous) [inline src/activities/CategoriesEntity.js:152]: Action is always generated now.

File: src/activities/CategoriesEntity.js (hunk 2 above)
Comment (mdulat) [reply src/activities/CategoriesEntity.js:178]: Do we need both `_generateSelectCategoryAction` and `_generateDeletePreviousAndAssociateWithNewCategory`? Wouldn't one action with the `shouldDeletePreviousCategory` suffice? The value of the field determines whether we delete the previous category or not?

Comment (mostlyfabulous) [reply src/activities/CategoriesEntity.js:178]: `_generateSelectCategoryAction` is needed for Assignments, were you thinking of refactoring this action to use the `shouldDeletePreviousCategory` param?

**Review by dustxd — APPROVED** [review]

File: src/activities/CategoriesEntity.js (hunk 2 above)
Comment (mdulat) [reply src/activities/CategoriesEntity.js:178]: Oh I see, no that makes sense, let's keep it separate then.

**Review by mdulat — APPROVED** [review]

**Pull request merged**